engine/jobmaster/dm/worker_manager.go

```diff
@@ -280,6 +283,29 @@ func (wm *WorkerManager) checkAndScheduleWorkers(ctx context.Context, job *metad
 			recordError = err
 			continue
 		}
+
+		// create success, record unit state
+		if err = wm.unitStore.ReadModifyWrite(ctx, func(state *metadata.UnitState) error {
+			status, ok := state.CurrentUnitStatus[taskID]
+			if !ok {
+				state.CurrentUnitStatus[taskID] = &metadata.TaskStatus{
+					Unit:        nextUnit,
+					Task:        taskID,
+					CreatedTime: time.Now(),
+					// we don't care about the Stage and CfgModRevision here
+				}
+			} else {
+				if status.Unit != nextUnit {
+					status.CreatedTime = time.Now()
+					status.Unit = nextUnit
+				}
+			}
+			return nil
+		}); err != nil {
+			wm.logger.Error("update current unit state failed", zap.String("task", taskID), zap.Error(err))
+			recordError = err
+			continue
```

Comment:
Any unit test for this part?

Reply:
added an unit test https://github.com/pingcap/tiflow/pull/7507/commits/54150a2001a9f245645c3238432d083cb84f6e8f